### PR TITLE
[Merged by Bors] - Fix: Port 8080 missing from Service in case tls is disabled

### PR DESCRIPTION
**Pull request by maltesander**
# Description

- If client TLS is deactivated the http port now is exposed correctly in the service (and container ports)
- Added test to check the service

fixes https://github.com/stackabletech/trino-operator/issues/299

test: https://ci.stackable.tech/view/02%20Operator%20Tests%20(custom)/job/trino-operator-it-custom/29/

## Review Checklist

- [x] Code contains useful comments
- [x] CRD change approved (or not applicable)
- [x] (Integration-)Test cases added (or not applicable)
- [x] Documentation added (or not applicable)
- [x] Changelog updated (or not applicable)
- [x] Cargo.toml only contains references to git tags (not specific commits or branches)
- [x] Helm chart can be installed and deployed operator works (or not applicable)

Once the review is done, comment `bors r+` (or `bors merge`) to merge. [Further information](https://bors.tech/documentation/getting-started/#reviewing-pull-requests)

### CODE REVIEW

File: rust/crd/src/lib.rs [hunk 1]
```diff
@@ -623,14 +623,14 @@ impl TrinoCluster {
         spec.config.internal_tls.as_ref()
     }
 
-    /// Returns if the HTTP port should be enabled
-    pub fn http_port_enabled(&self) -> bool {
-        self.get_client_tls().is_none() && self.get_internal_tls().is_none()
+    /// Returns if the HTTP port should be expose
```
Comment (sbernauer) [inline rust/crd/src/lib.rs:626]:
```suggestion
    /// Returns if the HTTP port should be exposed
```

**Review by sbernauer — APPROVED** [review]

File: rust/operator-binary/src/controller.rs [hunk 2]
```diff
@@ -1025,7 +1025,7 @@ fn container_ports(trino: &TrinoCluster) -> Vec<ContainerPort> {
 }
 
 fn readiness_probe(trino: &TrinoCluster) -> Probe {
-    let port_name = if trino.https_port_enabled() {
+    let port_name = if trino.get_client_tls().is_some() {
```
Comment (teozkr) [inline rust/operator-binary/src/controller.rs:1028]: How come these don't use `expose_https_port` now that they are equivalent?

Comment (maltesander) [reply rust/operator-binary/src/controller.rs]: done

File: rust/crd/src/lib.rs [hunk 3]
```diff
@@ -84,7 +84,7 @@ pub const LDAP_BIND_PASSWORD: &str = "ldap.bind-password";
 pub const LDAP_USER_BASE_DN: &str = "ldap.user-base-dn";
 pub const LDAP_GROUP_AUTH_PATTERN: &str = "ldap.group-auth-pattern";
 pub const LDAP_ALLOW_INSECURE: &str = "ldap.allow-insecure";
-pub const LDAP_SSL_TRUST_CERTIFICATE: &str = "ldap.ssl-trust-certificate";
+pub const LDAP_SSL_TRUST_STORE_PATH: &str = "ldap.ssl.truststore.path";
```
Comment (teozkr) [inline rust/crd/src/lib.rs:87]: 1. Did the old property break anything?
2. Looks like this was added in [Trino 376](https://trino.io/docs/current/release/release-376.html?highlight=ldap%20ssl%20truststore%20path), do we still support any versions older than that?

**Review by teozkr — APPROVED** [review]
LGTM

File: rust/crd/src/lib.rs (hunk 3 above)
Comment (maltesander) [reply rust/crd/src/lib.rs:87]: Yeah was added in 376 (our oldest is 377). In the doc examples (even for the newer versions like 395), they used the trust certificate but this broke 377 (no problem with newer versions strangely...).

Comment (maltesander) [reply rust/crd/src/lib.rs:87]: I did not investigate further but will start another custom test once the actions are done.

Comment (teozkr) [reply rust/crd/src/lib.rs:87]: Ok, makes sense then.

**Review by teozkr — APPROVED** [review]

**Comment by maltesander** [comment]
bors merge

**Pull request closed without merging**